### PR TITLE
[PMK-2205] Edit Cluster Page

### PR DESCRIPTION
**Pull request by xantorres**

### CODE REVIEW

File: src/app/plugins/kubernetes/components/infrastructure/clusters/EditClusterPage.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  const clusterId = match.params.id
+  const onComplete = useCallback(
+    success => success && history.push(listUrl),
+    [history])
+  const [clusters, loadingClusters] = useDataLoader(clusterActions.list)
+  const initialValues = useMemo(
+    () => pipe(
+      find(propEq('uuid', clusterId)),
+      when(isNil, always(emptyObj)),
+      ({ tags = {}, ...cluster }) => ({
```
Comment (brennancheung) [inline src/app/plugins/kubernetes/components/infrastructure/clusters/EditClusterPage.js]: Might want to move the tag part into its own function.

File: src/app/plugins/kubernetes/components/infrastructure/clusters/EditClusterPage.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        ...cluster,
+        tags: Object.entries(tags).map(([key, value]) => ({ key, value })),
+      }),
+    )(clusters),
+    [clusters, clusterId])
+  const [update, updating] = useDataUpdater(clusterActions.update, onComplete)
+  const handleSubmit = useCallback(({ tags, ...cluster }) => update({
+    ...cluster,
+    tags: tags.reduce((acc, { key, value }) => ({ ...acc, [key]: value }), {}),
```
Comment (brennancheung) [inline src/app/plugins/kubernetes/components/infrastructure/clusters/EditClusterPage.js]: Hard to see what this is doing at a glance.  Move into its own function and give it a good name that makes it self-documenting.

File: src/app/plugins/kubernetes/components/infrastructure/clusters/actions.js [hunk 3]
```diff
@@ -180,6 +176,14 @@ export const clusterActions = createCRUDActions(clustersCacheKey, {
       }
     }, rawClusters)
   },
+  createFn: (params, _, loadFromContext) => {
```
Comment (brennancheung) [inline src/app/plugins/kubernetes/components/infrastructure/clusters/actions.js]: Are we putting `createFn` below `listFn` normally?  Looks like just moving it around.

Comment (xantorres) [reply src/app/plugins/kubernetes/components/infrastructure/clusters/actions.js]: I just wanted to put together the edit operations (create/update/delete) as they are usually share some commonalities.

**Pull request merged**